### ISSUE TITLE
Readme undersells difficulty of using with TypeScript

### ISSUE THREAD

**Issue by lbv-stripe** [markdoc/markdoc#260]
### What happened?

The readme suggests that using with TypeScript is as simple as installing a few files and setting up a `tsconfig.json` file. But there are some pretty steep roadblocks later on. The particular one I've hit is that for the `config` object you pass to `Markdoc.transform`, the compiler expects a very complicated type — per `source/types.ts`, it's 

```
 type ConfigType = Partial<{
  nodes: Partial<Record<NodeType, Schema>>;
  tags: Record<string, Schema>;
  variables: Record<string, any>;
  functions: Record<string, ConfigFunction>;
  partials: Record<string, any>;
  validation?: {
    validateFunctions?: boolean;
  };
}>;
```
which is not something I know how to construct. The example `config` object given in the docs, 
```
const config = {
  tags: {
    callout
  }
};
```
fails to typecheck. The docs give no guidance, and as a result, I can't use any of Markdoc's interesting features. I'd appreciate either an example in the docs of a config object that will typecheck or (if all else fails) a heads-up in the readme that TS wizardry is required. 

### To reproduce

I encountered this following the instructions here: https://markdoc.dev/docs/tags

### Version

0.1.13

### Additional context

This may not count as a full-fledged bug, and probably merits the "docs" label. Sorry for the confusion, this was my only option. :)

**Comment by mskri** [comment]
I agree. It's been a trial and error at times to figure out how to type things so that TypeScript is happy.

**Comment by bwnelb** [comment]
I also found the type check errors from examples to be jarring and instead of learning a Markdoc concept I started learning some TS instead.  Might be good to note the `"strict": false,` tsconfig.json option if anyone wants a quick way to move past a type definition issue while exploring a new tool.

**Comment by JPrisk** [comment]
when working with typescript, adding the following to your config saves some guess work:
```ts
import Markdoc, { Config } from "@markdoc/markdoc";

const config: Config = {
	...
}
```

i'm reluctant to add this in a PR for the docs, because its primarily written for `js` atm

**Comment by fieldju** [comment]
> when working with typescript adding the following to your config saves some guess work:
> 
> ```ts
> import Markdoc, { Config } from "@markdoc/markdoc";
> 
> const config: Config = {
> 	...
> }
> ```
> 
> i'm reluctant to add this in a PR for the docs because its primarily written for `js` atm

FWIW for JS purists, you can add this annotation to get intellisense and IDE errors (vs transpilation errors)

`/** @type {import('@markdoc/markdoc').Config} */`

So, in this case it'd look something like

```javascript
/** @type {import('@markdoc/markdoc').Config} */
const config = {
	...
}
```

Which would give you give you IDE warning/errors and intellisense

![image](https://github.com/markdoc/markdoc/assets/711726/96e0265e-4e42-4e16-8b20-e9c325f4b360)

![image](https://github.com/markdoc/markdoc/assets/711726/5b085828-9e72-4b08-bb79-36092afa6aeb)

![image](https://github.com/markdoc/markdoc/assets/711726/541e9878-a07d-4921-84c8-8144f98c1c0b)

You can cmd+click into it to see the types too

![image](https://github.com/markdoc/markdoc/assets/711726/d0a3a9d2-c402-484c-9949-dfdd96b21388)

**Comment by JPrisk** [comment]
ah, thats a good suggestion @fieldju - have put up a PR for the team to consider
